strophe.h

```diff
@@ -397,7 +398,7 @@ void xmpp_stop(xmpp_ctx_t *ctx);
 
 /* UUID */
 
-char *xmpp_uuid_gen(xmpp_ctx_t *ctx);
+char *xmpp_uuid_gen(const xmpp_ctx_t * const ctx);
```

Comment:
`xmpp_uuid_get()` changes `xmpp_ctx_t`. Therefore, it must not be const. UUID generation uses `xmpp_rand_t` which changes its internal state on every generation.
Moreover, I would not merge the const patch. Ctx is controlled only by libstrophe and can be changed anywhere implicitly. For example, `xmpp_alloc()` may start storing statistics inside ctx.